app/utils/key-camelizer.js

```diff
@@ -0,0 +1,56 @@
+'use strict'
+
+function convertToCamelCase (objectKey) {
+  const camelizedString = objectKey.replace(/[-_\s]+(.)?/g, (match, chr) => chr ? chr.toUpperCase() : '')
+  return camelizedString.substr(0, 1).toLowerCase() + camelizedString.substr(1)
+}
+
+function separateWords (objectKey) {
+  const options = {}
+  const separator = options.separator || '_'
+  const split = options.split || /(?=[A-Z])/
+  return objectKey.split(split).join(separator).toLowerCase()
+};
+
+function camelize (obj) {
+  if (Array.isArray(obj)) {
+    return obj.map(item => camelize(item))
+  } else if (typeof obj === 'object') {
+    return Object.keys(obj).reduce((result, key) => {
+      if (obj[key] === null) {
+        result[convertToCamelCase(key)] = null
+      } else if (typeof obj[key] === 'object') {
+        result[convertToCamelCase(key)] = camelize(obj[key])
+      } else {
+        result[convertToCamelCase(key)] = obj[key]
+      }
+      return result
+    }, {})
+  } else {
+    return obj
+  }
+}
+
+function decamelize (obj) {
```

Comment:
Considering the above comment, it looks like `decamelize` only ever converts to snake case, sticking to KISS principles we should rename this function for clarity of purpose and remove the configuration options in `separateWords`.

Reply:
Good shout. Have renamed to `keysToSnakeCase`.